packages/fsshopify/src/mixins/ShopifyAccountDataSource.ts

```diff
@@ -0,0 +1,364 @@
+import { ShopifyAddress } from './../util/ShopifyResponseTypes';
+import { CommerceTypes } from '@brandingbrand/fscommerce';
+import { ShopifyMailingAddressInput } from '..';
+import DataSourceBase from '../util/DataSourceBase';
+import * as Normalizers from './../normalizers';
+
+export interface AddressResponse {
+  defaultAddress: string;
+  addresses: ShopifyAddress[];
+}
+
+export interface NormalizedAddressResponse {
+  defaultAddress: string;
+  addresses: CommerceTypes.Address[];
+}
+
+export interface ResponseError {
+  field: string | null;
+  message: string;
+}
+
+export interface ShopifyUserToken {
+  customerAccessTokenCreate: {
+    customerUserErrors: ResponseError[];
+    customerAccessToken: {
+      accessToken: string;
+      expiresAt: string;
+    };
+  };
+}
+
+export interface Customer {
+  customer: {
+    id: string;
+    firstName: string;
+    lastName: string;
+    email: string;
+    password?: string;
+  };
+  customerUserErrors?: {
+    message: string;
+  }[];
+  customerUpdate?: {
+    userErrors?: {
+      message: string;
+    }[];
+  };
+}
+
+export interface CustomerCreateResponse {
+  customerCreate: Customer;
+}
+
+export interface AddressCreateResponse {
+  customerAddressCreate: {
+    customerAddress: {
+      id: string;
+    };
+    userErrors: object[];
+  };
+}
+
+export interface AddressUpdateResponse {
+  customerAddressUpdate: {
+    customerAddress: {
+      id: string;
+    };
+    userErrors: object[];
+  };
+}
+
+export class ShopifyAccountDataSource extends DataSourceBase {
+
+  async getCustomerId(
+    customerAccessToken: string
+  ): Promise<Customer> {
+    const response = await this.api.postQuery(`
+      query($customerAccessToken: String!) {
+        customer (customerAccessToken: $customerAccessToken){
+          id
+          firstName
+          lastName
+          email
+        }
+      }`, {
+        customerAccessToken
+      });
+
+    return response;
+  }
+
+  async getAddresses(customerAccessToken: string): Promise<AddressResponse> {
+    const response = await this.api.postQuery(`
+    query ($customerAccessToken: String!) {
+      customer(customerAccessToken: $customerAccessToken) {
+        defaultAddress {
+          id
+        }
+        addresses(first: 20) {
+          edges {
+            node {
+              formatted
+              name
+              address1
+              address2
+              city
+              company
+              countryCodeV2
+              firstName
+              lastName
+              id
+              phone
+              province
+              provinceCode
+              zip
+            }
+          }
+        }
+      }
+    }`, {
+      customerAccessToken
+    });
+    const addresses = response.customer.addresses.edges.map((address: any) => {
```

Comment:
We should make sure this exists before accessing it. Same with defaultAddress below.